### ISSUE TITLE
"Module not compiled" for all modules in aeson

### ISSUE THREAD

**Issue by bergmark** [ndmitchell/weeder#3]
Here's a reproduction for aeson master, am I doing something wrong? It looks like the mentioned modules are being compiled

```
$ ln -s stack-lts8.yaml stack.yaml
$ weeder . --build
aeson-1.2.0.0: configure (lib + test)
aeson-1.2.0.0: build (lib + test)
aeson-1.2.0.0: copy/register
aeson-1.2.0.0: Test running disabled by --no-run-tests flag.
Completed 2 action(s).

--  Dumping log file due to warnings: /Users/adam.bergmark/repos/hs/aeson/.stack-work/logs/aeson-1.2.0.0.log

Configuring aeson-1.2.0.0...
Preprocessing library aeson-1.2.0.0...
[ 1 of 23] Compiling Data.Attoparsec.Time.Internal ( attoparsec-iso8601/Data/Attoparsec/Time/Internal.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Attoparsec/Time/Internal.o )
[ 2 of 23] Compiling Data.Attoparsec.Time ( attoparsec-iso8601/Data/Attoparsec/Time.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Attoparsec/Time.o )
[ 3 of 23] Compiling Data.Aeson.Types.Internal ( Data/Aeson/Types/Internal.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Types/Internal.o )
[ 4 of 23] Compiling Data.Aeson.Types.Generic ( Data/Aeson/Types/Generic.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Types/Generic.o )
[ 5 of 23] Compiling Data.Aeson.Parser.UnescapePure ( pure/Data/Aeson/Parser/UnescapePure.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Parser/UnescapePure.o )
[ 6 of 23] Compiling Data.Aeson.Parser.Unescape ( Data/Aeson/Parser/Unescape.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Parser/Unescape.o )
[ 7 of 23] Compiling Data.Aeson.Parser.Time ( Data/Aeson/Parser/Time.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Parser/Time.o )
[ 8 of 23] Compiling Data.Aeson.Parser.Internal ( Data/Aeson/Parser/Internal.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Parser/Internal.o )
[ 9 of 23] Compiling Data.Aeson.Parser ( Data/Aeson/Parser.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Parser.o )
[10 of 23] Compiling Data.Aeson.Internal.Time ( Data/Aeson/Internal/Time.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Internal/Time.o )
[11 of 23] Compiling Data.Aeson.Internal.Functions ( Data/Aeson/Internal/Functions.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Internal/Functions.o )
[12 of 23] Compiling Data.Aeson.Types.FromJSON ( Data/Aeson/Types/FromJSON.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Types/FromJSON.o )
[13 of 23] Compiling Data.Aeson.Internal ( Data/Aeson/Internal.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Internal.o )
[14 of 23] Compiling Data.Aeson.Encoding.Builder ( Data/Aeson/Encoding/Builder.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Encoding/Builder.o )
[15 of 23] Compiling Data.Aeson.Encoding.Internal ( Data/Aeson/Encoding/Internal.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Encoding/Internal.o )
[16 of 23] Compiling Data.Aeson.Encoding ( Data/Aeson/Encoding.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Encoding.o )
[17 of 23] Compiling Data.Aeson.Types.ToJSON ( Data/Aeson/Types/ToJSON.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Types/ToJSON.o )
[18 of 23] Compiling Data.Aeson.Types.Class ( Data/Aeson/Types/Class.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Types/Class.o )
[19 of 23] Compiling Data.Aeson.Types ( Data/Aeson/Types.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Types.o )
[20 of 23] Compiling Data.Aeson.Text  ( Data/Aeson/Text.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Text.o )
[21 of 23] Compiling Data.Aeson       ( Data/Aeson.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson.o )
[22 of 23] Compiling Data.Aeson.Encode ( Data/Aeson/Encode.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/Encode.o )
[23 of 23] Compiling Data.Aeson.TH    ( Data/Aeson/TH.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/Data/Aeson/TH.o )
Preprocessing test suite 'tests' for aeson-1.2.0.0...
[ 1 of 17] Compiling Types            ( tests/Types.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/Types.o ) [Data.Aeson.Types changed]
[ 2 of 17] Compiling Options          ( tests/Options.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/Options.o ) [Data.Aeson.Types changed]
[ 3 of 17] Compiling Functions        ( tests/Functions.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/Functions.o ) [Prelude.Compat changed]
[ 4 of 17] Compiling Instances        ( tests/Instances.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/Instances.o ) [Data.Hashable.Time changed]
[ 5 of 17] Compiling SerializationFormatSpec ( tests/SerializationFormatSpec.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/SerializationFormatSpec.o ) [Data.UUID.Types changed]
[ 6 of 17] Compiling ErrorMessages    ( tests/ErrorMessages.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/ErrorMessages.o ) [Data.HashMap.Strict changed]
[ 7 of 17] Compiling Encoders         ( tests/Encoders.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/Encoders.o ) [Data.Aeson.Types changed]
[ 8 of 17] Compiling Properties       ( tests/Properties.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/Properties.o ) [Data.UUID.Types changed]
[ 9 of 17] Compiling UnitTests.NullaryConstructors ( tests/UnitTests/NullaryConstructors.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/UnitTests/NullaryConstructors.o ) [Data.Aeson.Types changed]
[10 of 17] Compiling DataFamilies.Types ( tests/DataFamilies/Types.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/DataFamilies/Types.o ) [Generics.Deriving.TH changed]
[11 of 17] Compiling DataFamilies.Instances ( tests/DataFamilies/Instances.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/DataFamilies/Instances.o ) [Data.Aeson.Types changed]
[12 of 17] Compiling DataFamilies.Encoders ( tests/DataFamilies/Encoders.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/DataFamilies/Encoders.o ) [Data.Aeson.Types changed]
[13 of 17] Compiling DataFamilies.Properties ( tests/DataFamilies/Properties.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/DataFamilies/Properties.o ) [Prelude.Compat changed]
[16 of 17] Compiling UnitTests        ( tests/UnitTests.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/UnitTests.o ) [Data.HashSet changed]

/Users/adam.bergmark/repos/hs/aeson/tests/UnitTests.hs:501:47: warning: [-Wunused-top-binds]
    Defined but not used: ‘smf’
[17 of 17] Compiling Main             ( tests/Tests.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests-tmp/Main.o ) [Prelude.Compat changed]
Linking .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/tests/tests ...
Installing library in
/Users/adam.bergmark/repos/hs/aeson/.stack-work/install/x86_64-osx/lts-8.1/8.0.2/lib/x86_64-osx-ghc-8.0.2/aeson-1.2.0.0-J6hepn47qr4IYivjVNPua
Registering aeson-1.2.0.0...

--  End of log file: /Users/adam.bergmark/repos/hs/aeson/.stack-work/logs/aeson-1.2.0.0.log

Log files have been written to: /Users/adam.bergmark/repos/hs/aeson/.stack-work/logs/
= Package attoparsec-iso8601 =
No warnings

= Package aeson =

== Section library ==
Module not compiled
* Data.Aeson
* Data.Aeson.Encode
* Data.Aeson.Encoding
* Data.Aeson.Encoding.Builder
* Data.Aeson.Encoding.Internal
* Data.Aeson.Internal
* Data.Aeson.Internal.Functions
* Data.Aeson.Internal.Time
* Data.Aeson.Parser
* Data.Aeson.Parser.Internal
* Data.Aeson.Parser.Time
* Data.Aeson.Parser.Unescape
* Data.Aeson.TH
* Data.Aeson.Text
* Data.Aeson.Types
* Data.Aeson.Types.Class
* Data.Aeson.Types.FromJSON
* Data.Aeson.Types.Generic
* Data.Aeson.Types.Internal
* Data.Aeson.Types.ToJSON
* Data.Attoparsec.Time
* Data.Attoparsec.Time.Internal
Redundant build-depends entry
* attoparsec
* base
* base-compat
* containers
* deepseq
* dlist
* ghc-prim
* hashable
* scientific
* tagged
* template-haskell
* text
* time
* time-locale-compat
* unordered-containers
* uuid-types
* vector

== Section test:tests ==
Module not compiled
* Data.Aeson.Parser.UnescapeFFI
* Data.Aeson.Parser.UnescapePure
* DataFamilies.Encoders
* DataFamilies.Instances
* DataFamilies.Properties
* DataFamilies.Types
* Encoders
* ErrorMessages
* Functions
* Instances
* Options
* Properties
* SerializationFormatSpec
* Tests.hs
* Types
* UnitTests
* UnitTests.NullaryConstructors
Redundant build-depends entry
* HUnit
* QuickCheck
* aeson
* attoparsec
* base
* base-compat
* base-orphans
* base16-bytestring
* containers
* directory
* dlist
* filepath
* generic-deriving
* ghc-prim
* hashable
* integer-logarithms
* quickcheck-instances
* scientific
* tagged
* template-haskell
* test-framework
* test-framework-hunit
* test-framework-quickcheck2
* text
* time
* time-locale-compat
* unordered-containers
* uuid-types
* vector
```

**Comment by bergmark** [comment]
This seems like a great tool by the way, thanks a lot for making it!

**Comment by ndmitchell** [comment]
TIL you can have a space separated hs-source-dirs. Solution now checked in, and just about to make a 0.1.2 release. For your info the aeson output is now below. I'd welcome any feedback as to whether it's right, wrong or helpful.

```
= Package aeson =

== Section library ==
Excessive other-modules entry
* Data.Aeson.Parser.Unescape
Missing other-modules entry
* Data.Aeson.Parser.UnescapePure
Weeds exported
* Data.Aeson.Encoding.Builder
  - array
  - ascii2
  - ascii4
  - ascii5
  - bool
  - object
* Data.Aeson.Parser.Time
  - timeZone
* Data.Aeson.Types.Generic
  - And
* Data.Aeson.Types.ToJSON
  - gToJSON

== Section test:tests ==
Excessive other-modules entry
* Data.Aeson.Parser.UnescapeFFI
* Data.Aeson.Parser.UnescapePure
* DataFamilies.Instances
* Functions
* Instances
Weeds exported
* Data.Aeson.Parser.UnescapeFFI
  - unescapeText
* Data.Aeson.Parser.UnescapePure
  - unescapeText
* DataFamilies.Encoders
  - thGADTToEncodingDefault
  - thGADTToEncodingUnwrap
* DataFamilies.Types
  - C4
* Encoders
  - gApproxParseJSONDefault
  - gApproxParseJSONUnwrap
  - gApproxToEncodingUnwrap
  - gApproxToJSONUnwrap
  - incoherentInstancesNeededParseJSONString
  - thGADTToEncodingDefault
  - thGADTToEncodingUnwrap
* Options
  - optsDefault
* Properties
  - ==~
  - P6
  - S4
  - checkValue
  - encodeDouble
  - encodeInteger
  - isEmptyArray
  - isEmptyArray'
  - isTaggedObject'
  - isTaggedObjectValue
  - isUntaggedValueETI
  - modifyFailureProp
  - roundTripEnc
  - roundTripEq
  - roundTripKey
  - roundTripNoEnc
  - sameAs1
  - sameAs1Agree
  - toFromJSON
  - toParseJSON1
* Types
  - zeroDay
```

**Comment by ndmitchell** [comment]
I just released weeder-0.1.2.

**Comment by bergmark** [comment]
Thanks a lot! I'll get back to you with more feedback.
